### PR TITLE
tests(fixtures): Make loadFixtures more generic

### PR DESCRIPTION
**Pull request by evanpurkhiser**
1) You can now load JSON and YAML fixtures
2) "flattening" of fixtures is now a configuration, otherwise you'll just get back a dict of your fixtures

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -9618,6 +9628,13 @@ js-yaml@^3.12.0, js-yaml@^3.13.1:
     argparse "^1.0.7"
     esprima "^4.0.0"
 
+js-yaml@^4.1.0:
```
Comment (EvanPurkhiser) [inline yarn.lock:9631]: damn, we have this as a transient dep, but the transient version is 3.x 😕

**Comment by billyvg** [comment]
I think json would be better —
* it's more widely used and supported
* there's native parsing for it
* we don't need any of the complex features of yaml (e.g. anchors/aliases) - I would assume parsing JSON would be faster too
* readability is a bit subjective, but braces are easier to navigate/match up over whitespaces (e.g. using `%` in vim to go to the matching brace).

**Comment by EvanPurkhiser** [comment]
Yeah I won't argue super hard here, removed yaml

File: package.json [hunk 2]
```diff
@@ -147,6 +147,7 @@
     "@storybook/builder-webpack5": "^6.2.9",
     "@storybook/react": "^6.2.9",
     "@storybook/theming": "^6.2.9",
+    "@types/js-yaml": "^4.0.1",
```
Comment (billyvg) [inline package.json:150]: Is this still needed?

**Review by taylangocmen — APPROVED** [review]
lgtm

**Review by billyvg — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (EvanPurkhiser) [reply package.json:150]: oops

**Pull request merged**